### PR TITLE
Add Slack Messaging (using eng-pipes)

### PR DESCRIPTION
**Pull request by brian-lou**

### CODE REVIEW

File: infra_event_notifier/backends/slack.py [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+    Sends an event to Slack via eng-pipes.
+    See: https://github.com/getsentry/eng-pipes
+
+    :param title: Title of Slack Message
+    :param text: Body of Slack Message
+    :param channel_id: ID of the Slack channel to send the message to
+    :param eng_pipes_key: Secret Key used to HMAC sign request
+    :param eng_pipes_url: Full URL for eng-pipes slack webhooks
+    """
+    # API docs: https://docs.datadoghq.com/api/latest/events/#post-an-event
```
Comment (bmckerry) [inline infra_event_notifier/backends/slack.py]:
```suggestion
```

File: infra_event_notifier/backends/slack.py [hunk 2]
```diff
@@ -1,3 +1,51 @@
...
+    payload = {
+        "source": "kafka-control-plane",
+        "title": title,
+        "body": text,
+        "channel": channel_id,
```
Comment (bmckerry) [inline infra_event_notifier/backends/slack.py]: `channel_id` was removed from the eng-pipes webhook, we just use a single channel hardcoded in eng-pipes for now

File: infra_event_notifier/slack_notifier.py [hunk 3]
```diff
@@ -2,15 +2,32 @@
...
+        Sends the message to slack
         """
-        # TODO: implement
-        if self.slack_api_key is not None and title and body:
-            send_notification(
-                title=title, text=body, slack_api_key=self.slack_api_key
-            )
+        assert self.eng_pipes_key is not None, "Missing Eng-Pipes Signing Key"
+        assert self.eng_pipes_url is not None, "Missing Eng-Pipes API URL"
+        assert channel_id is not None, "Missing Slack Channel ID"
```
Comment (bmckerry) [inline infra_event_notifier/slack_notifier.py]: same thing here, should remove `channel_id`

File: tests/backends/test_slack.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+from unittest.mock import MagicMock, patch
+from urllib.error import HTTPError
+
+import pytest
+
+from infra_event_notifier.backends.slack import send_notification
+
+
+def mock_context_manager() -> MagicMock:
+    # mocking context managers is hard
```
Comment (victoria-yining-huang) [inline tests/backends/test_slack.py:11]: is this comment added intentionally?

**Review by victoria-yining-huang — APPROVED** [review]

**Review by victoria-yining-huang — APPROVED** [review]

**Pull request merged**